src/main/java/de/tum/in/www1/artemis/web/rest/AthenaResource.java

```diff
@@ -1,81 +1,208 @@
+        log.debug("REST call to get student repository for exercise {}, submission {}", exerciseId, submissionId);
+        checkAthenaSecret(auth);
+        return ResponseUtil.ok(athenaRepositoryExportService.exportRepository(exerciseId, submissionId, null));
+    }
+
+    /**
+     * GET public/athena/programming-exercises/:exerciseId/repository/template : Get the template repository as a zip file download
+     *
+     * @param exerciseId the id of the exercise
+     * @param auth       the auth header value to check
+     * @return 200 Ok with the zip file as body if successful
+     */
+    @GetMapping("public/athena/programming-exercises/{exerciseId}/repository/template")
+    @EnforceNothing // We check the Athena secret instead
+    @ManualConfig
+    public ResponseEntity<Resource> getTemplateRepository(@PathVariable long exerciseId, @RequestHeader("Authorization") String auth) throws IOException {
+        log.debug("REST call to get template repository for exercise {}", exerciseId);
+        checkAthenaSecret(auth);
+        return ResponseUtil.ok(athenaRepositoryExportService.exportRepository(exerciseId, null, RepositoryType.TEMPLATE));
+    }
+
+    /**
+     * GET public/athena/programming-exercises/:exerciseId/repository/solution : Get the solution repository as a zip file download
+     *
+     * @param exerciseId the id of the exercise
+     * @param auth       the auth header value to check
+     * @return 200 Ok with the zip file as body if successful
+     */
+    @GetMapping("public/athena/programming-exercises/{exerciseId}/repository/solution")
+    @EnforceNothing // We check the Athena secret instead
+    @ManualConfig
+    public ResponseEntity<Resource> getSolutionRepository(@PathVariable long exerciseId, @RequestHeader("Authorization") String auth) throws IOException {
+        log.debug("REST call to get solution repository for exercise {}", exerciseId);
+        checkAthenaSecret(auth);
+        return ResponseUtil.ok(athenaRepositoryExportService.exportRepository(exerciseId, null, RepositoryType.SOLUTION));
+    }
+
+    /**
+     * GET public/athena/programming-exercises/:exerciseId/repository/tests : Get the test repository as a zip file download
+     *
+     * @param exerciseId the id of the exercise
+     * @param auth       the auth header value to check
+     * @return 200 Ok with the zip file as body if successful
+     */
+    @GetMapping("public/athena/programming-exercises/{exerciseId}/repository/tests")
+    @EnforceNothing // We check the Athena secret instead
+    @ManualConfig
+    public ResponseEntity<Resource> getTestRepository(@PathVariable long exerciseId, @RequestHeader("Authorization") String auth) throws IOException {
+        log.debug("REST call to get test repository for exercise {}", exerciseId);
+        checkAthenaSecret(auth);
+        return ResponseUtil.ok(athenaRepositoryExportService.exportRepository(exerciseId, null, RepositoryType.TESTS));
```

Comment:
why do we need responseutil?
can we not use either `ResponseEntity.ok(body)` or remove the ResponseEntity entirely and directly return a Resource.
Also applies to the other cases where ResponseUtil is used in this file.

Reply:
The `exportRepository` method returns a file. When I try to return the file directly (or with a `ResponseEntity.ok(...)`, I only get the file path as the response. The util creates a download stream and sets the necessary headers.